tests/projects/test_project.py

```diff
@@ -440,6 +440,34 @@ def test_set_function_with_relative_path(context):
     )
 
 
+@pytest.mark.parametrize(
+    "artifact_path,file_exists,expectation",
+    [
+        ("handler.py", True, does_not_raise()),
+        ("handler.py", False, pytest.raises(OSError)),
+    ],
+)
+def test_set_artifact_validates_file_exists(
+    monkeypatch, artifact_path, file_exists, expectation
+):
+    artifact_key = "my-artifact"
+    project = mlrun.new_project("inline", context=str(assets_path()), save=False)
+
+    monkeypatch.setattr(
```

Comment:
why `monkeypatch`? I believe you can just give 1 existing file and 1 that doesn't

Reply:
Indeed, but there's no need to create real files for this unit test. I don't care about their content, I'm just checking the "file exists" functionality